wgpu-core/src/lock/non_validating.rs

```diff
@@ -0,0 +1,37 @@
+/*! Methods that do not validate lock acquisition order. */
+#![allow(dead_code)]
```

Comment:
Ah, you're doing this because this might not be used when not validating. Is there even a reason to compile the modules when one is not selected? At the very least, it seems we can make this suppression use some meaningful `cfg` conditions. Perhaps in the parent module declaration we could use `#[cfg_attr(…, allow(dead_code))]` instead to keep the mirrored conditions closer together?